stan/math/rev/fun/conv_gaus_line.hpp

```diff
@@ -0,0 +1,81 @@
+#ifndef STAN_MATH_REV_FUN_CONV_GAUS_LINE
+#define STAN_MATH_REV_FUN_CONV_GAUS_LINE
+
+#include <stan/math/rev/meta.hpp>
+#include <stan/math/rev/core.hpp>
+#include <stan/math/prim/fun/conv_gaus_line.hpp>
+
+namespace stan {
+namespace math {
+
+namespace internal {
+
+/** \ingroup prob_dists
+ * Evaluates the derivative of the convolution of a line with a Gaussian
+ * kernel on an interval.
+ *
+ * \f$\frac{\partial}{\partial x} \int_{t_0}^{t_1} (at + b) \f$
+ * \f$ e^{\frac{-(t-x)^2}{2\sigma^2}} dt \f$
+ *
+ * @param t0 lower integration bound
+ * @param t1 upper integration bound
+ * @param a coefficient of t in line
+ * @param b constant in line
+ * @param x0 point at which convolution is evaluated
+ * @param sig2 variance of the Gaussian kernel
+ * @return The value of the derivative
+ */
+inline double der_conv_gaus_line(double t0, double t1, double a, double b,
+                                 double x0, double sig2) {
+  using stan::math::normal_cdf;
+  using std::exp;
+  using std::pow;
+  using std::sqrt;
+  const double pi = stan::math::pi();
+  const double sig = sqrt(sig2);
+  const double alpha = sqrt(2 * pi * sig2);
+
+  double y = (a * x0 + b) / alpha
+             * (-exp(-pow(t1 - x0, 2) / (2 * sig2))
+                + exp(-pow(t0 - x0, 2) / (2 * sig2)));
+  y += a * (normal_cdf(t1, x0, sig) - normal_cdf(t0, x0, sig));
+  y -= a / alpha
+       * ((t1 - x0) * exp(-pow(t1 - x0, 2) / (2 * sig2))
+          - (t0 - x0) * exp(-pow(t0 - x0, 2) / (2 * sig2)));
```

Comment:
[very optional] Wonder if there is someway to simplify the algebra here? `exp` and `pow` are pretty expensive

Reply:
Instead of doing `exp(-pow(t1 - x0, 2) / (2 * sig2)` multiple times would be good to pull out and do once